### PR TITLE
Added dynamic command handling, added perms and argument checks

### PR DESCRIPTION
**Pull request by ghostiek**

### CODE REVIEW

File: src/bot.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/*
+Here we gooooooooooo.
+*/
+
+// Imports
```
Comment (antoinegag) [inline src/bot.js]: Not needed

File: src/bot.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const fs = require('fs')
+const Discord = require('discord.js');
+
+// Grab config info from the secret config.
```
Comment (antoinegag) [inline src/bot.js]: Not needed

File: src/bot.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+const { token, prefix } = require('./auth/config.json');
+
+// Grab a new Discord client instance
```
Comment (antoinegag) [inline src/bot.js]: Not needed

File: src/bot.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+const client = new Discord.Client();
+
+//Initialize commands
```
Comment (antoinegag) [inline src/bot.js]: Not needed and wrong

File: src/bot.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));
+
+for (const file of commandFiles) {
+  const command = require(`./commands/${file}`);
+  client.commands.set(command.name, command);
+}
+
+// Ready player one
+// AKA waits for 'ready' event to fire, and then lets the world (e.g. the console) know.
```
Comment (antoinegag) [inline src/bot.js]: IMO this has no reason to be in code directly, you can figure that out easily if you're unexperienced with the API

File: src/bot.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+client.on('ready', () => {
+  console.log(`${client.user.tag} online... beep boop.`);
+});
+
+
+//Listening to messages to trigger commands
```
Comment (antoinegag) [inline src/bot.js]: Not needed, same as on ready

File: src/bot.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+client.on('message', message => {
+
+  //Ignores messages without prefix or bot messages
```
Comment (antoinegag) [inline src/bot.js]: Not needed, the condition in the if is clear enough

File: src/bot.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  if (!message.content.startsWith(prefix) || message.author.bot) return;
+
+  const args = message.content.slice(prefix.length).split(/ +/);
+  const commandName = args.shift().toLowerCase();
+
+  //Ignore invalid commands
```
Comment (antoinegag) [inline src/bot.js]: Not needed, the condition in the if is clear enough

File: src/bot.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  if (!client.commands.has(commandName)) return;
+
+  const command = client.commands.get(commandName);
+
+  //Verifies if the command was sent as a DM or was executed on the server
```
Comment (antoinegag) [inline src/bot.js]: Not needed, the condition in the if is clear enough

File: src/bot.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  if (command.guildOnly && message.channel.type !== 'text') {
+    return message.reply('I can\'t execute that command inside DMs!');
+  }
+
+  //Inform user that his command was incorrectly formatted
```
Comment (antoinegag) [inline src/bot.js]: Not needed, the condition in the if is clear enough

File: src/bot.js [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+  if (command.args && args.length < command.minArgs) {
+    let reply = `You didn't provide enough arguments, ${message.author}!`;
+
+    //Explains format if defined
```
Comment (antoinegag) [inline src/bot.js]: Not needed, it's just a null check

**Comment by CMCarlsen** [comment]
Went ahead and merged this. Still considering best practices for certain things, but will consider this our master for now.

**Pull request merged**